app/src/main/java/com/codesoom/assignment/controllers/ProductController.java

```diff
@@ -0,0 +1,55 @@
+package com.codesoom.assignment.controllers;
+import com.codesoom.assignment.applications.ProductService;
+import com.codesoom.assignment.domains.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import java.util.List;
+
+@RestController
+@RequestMapping("/products")
+@CrossOrigin
+public class ProductController {
+    @Autowired
+    ProductService ProductService;
+
+    public ProductController(ProductService ProductService) {
+        this.ProductService = ProductService;
+    }
```

Comment:
이 `@Autowired`는 생략해도 문제 없습니다. (바로 아래에 있는 생성자가 힌트입니다) 왜 그런지에 대해서 스프링 공식문서를 읽으며 조사해 보세요. 엄청 좋은 공부가 될 거에요.

```suggestion
    ProductService ProductService;

    public ProductController(ProductService ProductService) {
        this.ProductService = ProductService;
    }
```

Reply:
@Autowired 로 선언한 Component는 Test나 Application Run할 때 미리 등록해서 항상 같은 객체를 접근할 수있고,
아래의 생성자의 경우는  new를 통해 service를 새로 생성하지만 항상 controller와 짝이 지어지기 때문에 둘 다 동일한 결과를 얻을 수 있을 것같습니다.

@Autowired의 공식문서도 공부해보겠습니다. 자주활용하는데 아주 간단한 정보만 알고 사용하고있었네요